### PR TITLE
State transition manager test coverage

### PR DESCRIPTION
**Pull request by neotheprogramist**
## What ❔

This PR enhances the test coverage for the ```StateTransitionManager``` contract. Specifically, it includes additional unit tests to cover edge cases and scenarios that were previously untested.

### CODE REVIEW

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/AdminManagement.t.sol [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import {StateTransitionManagerTest} from "./_StateTransitionManager_Shared.t.sol";
+
+contract AdminManagement is StateTransitionManagerTest {
+    function test_RevertWhen_IsNotAdminOrOwner() public {
+        address newAdmin = makeAddr("newadmin");
```
Comment (vladbochok) [inline l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/AdminManagement.t.sol:8]: ?
```suggestion
        address newAdmin = makeAddr("newAdmin");
```

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/FreezeChain.t.sol [hunk 2]
```diff
@@ -26,4 +26,22 @@ contract freezeChainTest is StateTransitionManagerTest {
...
+        uint256 newChainid = 10;
+        createNewChainWithId(getDiamondCutData(diamondInit), newChainid);
+
+        address newChainAddress = chainContractAddress.getHyperchain(newChainid);
+        GettersFacet gettersFacet = GettersFacet(newChainAddress);
+        bool isChainFrozen = gettersFacet.isDiamondStorageFrozen();
+        assertEq(isChainFrozen, false);
+
+        vm.stopPrank();
+        vm.startPrank(governor);
```
Comment (vladbochok) [inline l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/FreezeChain.t.sol:40]:
```suggestion
        vm.prank(governor);
```

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol [hunk 3]
```diff
@@ -36,4 +36,53 @@ contract createNewChainTest is StateTransitionManagerTest {
...
+    function test_RevertWhen_HyperchainLimitReached() public {
+        for (uint256 i = 0; i < MAX_NUMBER_OF_HYPERCHAINS; i++) {
+            createNewChainWithId(getDiamondCutData(diamondInit), 10 + i);
+        }
+
+        uint256[] memory chainIds = chainContractAddress.getAllHyperchainChainIDs();
+        assertEq(chainIds.length, MAX_NUMBER_OF_HYPERCHAINS);
+
+        vm.stopPrank();
+        vm.startPrank(governor);
```
Comment (vladbochok) [inline l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol:83]: Why do you use this pattern?

Comment (tejks) [reply l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/CreateNewChain.t.sol]: I used this pattern because I did not want to interfere with the tests that were previously created by someone else. However, I made a small modification and now it should be fine.

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/AdminManagement.t.sol (hunk 1 above)
Comment (tejks) [reply l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/AdminManagement.t.sol]: Fixed

File: l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/FreezeChain.t.sol (hunk 2 above)
Comment (tejks) [reply l1-contracts/test/foundry/unit/concrete/state-transition/StateTransitionManager/FreezeChain.t.sol]: Fixed

**Comment by saxenism** [comment]
Closed in favor of #1251
